library/constant_time.c

```diff
@@ -78,11 +78,11 @@ static inline uint32_t mbedtls_get_unaligned_volatile_uint32(volatile const unsi
      */
     uint32_t r;
 #if defined(MBEDTLS_CT_ARM_ASM)
-    asm volatile ("ldr %0, [%1]" : "=r" (r) : "r" (p) :);
+    __asm volatile ("ldr %0, [%1]" : "=r" (r) : "r" (p) :);
```

Comment:
In `common.h` there is

```
#ifndef asm
#if defined(__IAR_SYSTEMS_ICC__)
#define asm __asm
```

which should take care of this, so can you fix the underlying problem rather than this one instance?

Reply:
Thanks, I see that has been fixed in the time between this patch being pushed onto tf-m until now. So I think I can just get rid of this here.

Reply:
Added a commit which removes this workaround from the original workaround. I decided to keep the original commit as well to better maintain the history (as this was submitted "as is" to TF-M originally).